### ISSUE TITLE
Feature: Add a comparison text box for hashes

### ISSUE THREAD

**Issue by Josh65-2201** [files-community/Files#14891]
### What feature or improvement do you think would benefit Files?

*

### Requirements

- Add a text input at the bottom of the hash page in properties
- When entered show matching hash type as text below text input.
   - Alternatively highlight match with green text and show as all red if there is no match

### Files Version

3.2.6.0

### Windows Version

10.0.22621.3155

**Comment by 0x5bfa** [comment]
I don't see usecase. is this for user to check if hashes from two files are the same?

**Comment by Josh65-2201** [comment]
No. Sites usually will have the hash for downloaded files, comparing a SHA256 yourself is slow since it so long and doesn't fit in the properties window with the default size making a character easily missed.

**Comment by yaira2** [comment]
We'll need a design for this

**Comment by 0x5bfa** [comment]
![image](https://github.com/files-community/Files/assets/62196528/19933592-67d9-4e43-b724-e721aa945a24)
From HashTab app

**Comment by yaira2** [comment]
@Lamparter there are two aspects to this feature request
1. A new section with a textbox where users can enter a hash, and Files will display if it's a match for any of the hashes.
2. A button to compare with another file to check they have the same hash.

I'll keep this under planning as we don't have a final design, but the functionality aspect can be started on.

**Comment by yaira2** [comment]
@Josh65-2201 this feature will be included in v4.

**Comment by Lamparter** [comment]
Amazing!
